evap/staff/views.py

```diff
@@ -1037,21 +1080,44 @@ def user_create(request):
 
 @staff_required
 def user_import(request):
-    form = UserImportForm(request.POST or None, request.FILES or None)
-    operation = request.POST.get('operation')
+    excel_form = UserImportForm(request.POST or None, request.FILES or None)
 
-    if form.is_valid():
+    errors = []
+    warnings = []
+    success_messages = []
+
+    if request.method == "POST":
+        operation = request.POST.get('operation')
         if operation not in ('test', 'import'):
             raise SuspiciousOperation("Invalid POST operation")
 
         test_run = operation == 'test'
-        excel_file = form.cleaned_data['excel_file']
-        UserImporter.process(request, excel_file, test_run)
+        import_run = operation == 'import'
+
         if test_run:
-            return render(request, "staff_user_import.html", dict(form=form))
-        return redirect('staff:user_index')
-    else:
-        return render(request, "staff_user_import.html", dict(form=form))
+            excel_form.excel_file_required = True
+            if excel_form.is_valid():
+                excel_file = excel_form.cleaned_data['excel_file']
+                file_content = excel_file.read()
+                __, success_messages, warnings, errors = UserImporter.process(file_content, test_run)
+                if not errors:
+                    save_import_file(excel_file, str(request.user.id))
+                else:
+                    delete_import_file(str(request.user.id))  # remove old file if still exists
```

Comment:
to me, it would make more sense to do the delete directly below the `if test_run:` line, because that's the place where we know we don't need the old file. what do you think?